### PR TITLE
Use p-map instead of promise-pool

### PR DESCRIPTION
**Pull request by firecow**
@naweiss Have a look. See https://github.com/firecow/gitlab-ci-local/discussions/943 for more information.

promise-pool package doesn't throw on promise rejections, and we want that to happen, so developers can see error messages.

### CODE REVIEW

**Comment by naweiss** [comment]
@firecow can you give a minimal example to check the issue against?

**Comment by firecow** [comment]
> @firecow can you give a minimal example to check the issue against?

Good idea, we don't want this to come back 👍

**Comment by firecow** [comment]
@naweiss I've added a test to prevent regression :+1:

File: src/executor.ts [hunk 1]
```diff
@@ -12,12 +12,8 @@ export class Executor {
         do {
             startCandidates = Executor.getStartCandidates(jobs, stages, potentialStarters, argv.manual);
             if (startCandidates.length > 0) {
-                await PromisePool
-                    .withConcurrency(argv.concurrency ?? startCandidates.length)
-                    .for(startCandidates)
```
Comment (naweiss) [inline src/executor.ts:17]: I looked into the documentation of "@supercharge/promise-pool", and does seems to support throwing on promise rejections by adding:
```
.handleError(async (error) => { throw error; })
```
So, we should decide which library will be better.

Comment (firecow) [reply src/executor.ts:17]: p-map is made by https://github.com/sindresorhus and has zero dependencies, it's fine 👍

**Review by naweiss — APPROVED** [review]

**Pull request merged**